internals/plan/plan.go

```diff
@@ -42,11 +68,90 @@ const (
 	defaultCheckThreshold = 3
 )
 
+var (
+	// sectionExtensions keeps a map of registered extensions.
+	sectionExtensions = map[string]SectionExtension{}
+
+	// sectionExtensionsOrder records the order in which the extensions were registered.
+	sectionExtensionsOrder = []string{}
```

Comment:
@benhoyt I experimented with a slice of structs to make ```sectionExtensions``` ordered, but reverted to just adding a separate slice for key ordering. I left it left the overall code in a simpler state, as ordering is only applicable at Marshal time, while map access simplifies the remainder of the code. Let me know what you think.

Reply:
OK from Ben. Resolving.